pkg/handler/metrics.go

```diff
@@ -68,43 +70,55 @@ func getTopologyFlows(cfg *loki.Config, client httpclient.Caller, params url.Val
 	}
 
 	merger := loki.NewMatrixMerger(reqLimit)
+	var queries []string
+
 	if len(filterGroups) > 1 {
 		// match any, and multiple filters => run in parallel then aggregate
-		var queries []string
 		for _, group := range filterGroups {
-			query, code, err := buildTopologyQuery(cfg, group, start, end, limit, metricFunction, metricType, reporter, layer, scope, groups)
+			query, code, err := buildMetricsQuery(cfg, group, start, end, limit, step, metricFunction, metricType, reporter, layer, scope, groups)
 			if err != nil {
 				return nil, code, errors.New("Can't build query: " + err.Error())
 			}
 			queries = append(queries, query)
+
+			appQuery, appCode, appErr := buildMetricsQuery(cfg, group, start, end, "1", step, metricFunction, metricType, reporter, layer, "app", "")
```

Comment:
Sure, I will split these.

My plan was to use it in query summary on both Overview & Topology tabs.
It could also fill the blanks on Table tab when limit is reached